src/main.ml

```diff
@@ -166,15 +170,22 @@ let main port xen_api_uri certfile ciphersuites no_tls =
                     (* ignore the exception resulting from double-closing the socket *)
                     (ignore_exn (fun () -> Lwt_unix.close fd))
                )
-               (fun e -> Lwt_io.eprintf "Caught %s\n%!" (Printexc.to_string e))
+               (fun e -> Lwt_log.error_f "Caught exception while handling client: %s" (Printexc.to_string e))
            in
            loop ()
          in
          loop ()
       )
       (ignore_exn (fun () -> Lwt_unix.close sock))
   in
-  Lwt_main.run t;
+  (* Log unexpected exceptions *)
+  Lwt_main.run
+    (Lwt.catch t
+       (fun e ->
+          Lwt_log.fatal_f "Caught unexpected exception: %s" (Printexc.to_string e) >>= fun () ->
```

Comment:
Apparently `fatal` will broadcast the message to all consoles, which may not be what we want:
```
Broadcast message from systemd-journald@dt14 (Wed 2017-09-13 17:49:33 UTC):
xapi-nbd[25973]: main: Caught unexpected exception: Ssl.Certificate_error
```
If it fails many times in a row during startup, then we will get many repeated broadcast messages in a short period of time. I'm happy to change this to `error` if this is too much noise or not what we should use.

Reply:
Fatal sounds scary! :grin:

Reply:
The definitions of the error and fatal levels, from the docs:
```
| Error: An error message, which should not means the end of the program
| Fatal: A fatal error happened, in most cases the program will end after a fatal error
```